sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs

```diff
@@ -149,7 +149,9 @@ private static UcfgLocation GetLocation(SyntaxNode syntaxNode)
         public static string GetMethodId(IMethodSymbol methodSymbol)
         {
             // Generic methods and methods declared in a base class
-            var originalSymbol = methodSymbol?.ReducedFrom ?? methodSymbol?.OriginalDefinition;
+            var originalSymbol = methodSymbol?.ExplicitInterfaceImplementations.FirstOrDefault()
```

Comment:
Don't bother extracting this into a separate method but please add a comment on top explaining why this order and why this is split into 3 checks. For example which kind of cases do you want to catch when checking `methodSymbol?.ReducedFrom`?